sympy/solvers/solveset.py

```diff
@@ -2808,8 +2808,8 @@ def linsolve(system, *symbols):
     * An error is raised if, after expansion, any nonlinearity
       is detected:
 
-    >>> linsolve([x*(1/x - 1), (y - 1)**2 - y**2 + 1], x, y)
-    FiniteSet((1, 1))
+    #>>> linsolve([x*(1/x - 1), (y - 1)**2 - y**2 + 1], x, y)
+    #FiniteSet((1, 1))
```

Comment:
The current implementation in this PR does not handle this case. It could be made to handle this case by using `expand` but that would slow things down in general.

I don't think it's reasonable that a solver for linear equations should be expected to cancel nonlinear terms like this. Unfortunately this is documented behaviour.

I would like to simply remove this functionality but that is not a backwards compatible change. I doubt that many users are making reasonable use of this feature but I'm sure that someone's code would break if it were removed.

Does anyone have any opinions on this?

Reply:
I restored the old behaviour. I think it should be deprecated but for now I'd just like to finish this and get it merged.